### PR TITLE
Volume ffmpeg source

### PR DESCRIPTION
**Pull request by EddyGharbi**

### CODE REVIEW

File: deps/media-playback/media-playback/media.c [hunk 1]
```diff
@@ -377,12 +377,21 @@ static void mp_media_next_audio(mp_media_t *m)
...
 			return;
 
-		audio = malloc(sizeof(struct obs_source_audio));
 
+		audio = malloc(sizeof(struct obs_source_audio));
 		for (size_t i = 0; i < MAX_AV_PLANES; i++) {
 			if (f->data[i]) {
 				audio->data[i] = malloc(f->linesize[0]);
-				memcpy((void*)audio->data[i], f->data[i], f->linesize[0]);
+				if (m->volume < 100) {
```
Comment (summeroff) [inline deps/media-playback/media-playback/media.c:385]: Better to compare to 100.0f to be sure float comparison works correctly.

Comment (EddyGharbi) [reply deps/media-playback/media-playback/media.c:385]: I had one left over of my initial implementation that was using floats for the volume. It's intended to compare to an integer.

Comment (EddyGharbi) [reply deps/media-playback/media-playback/media.c:385]: (good catch)

**Review by summeroff — APPROVED** [review]

File: deps/media-playback/media-playback/media.c (hunk 1 above)
Comment (summeroff) [reply deps/media-playback/media-playback/media.c:385]: but float was needed for a line below to work correctly. 

**out[j] = m->volume / 100 * in[j];**

**m->volume / 100** always be 0

Comment (EddyGharbi) [reply deps/media-playback/media-playback/media.c:385]: Updated, good catch

**Review by summeroff — APPROVED** [review]

**Pull request merged**